### PR TITLE
fix: do not hold genesis records in memory

### LINKED ISSUE #3314: Do not keep genesis records in memory
Now that https://github.com/near/near-explorer/issues/393 is closed, we can stop holding genesis records (which can be several gigabytes large) in memory to lower memory usage.

### PR DESCRIPTION
**Pull request by bowenwang1996**
Fixes #3314.

Test plan
---------
* CI tests
* test with a real node to see that memory usage decreases on node start.

### CODE REVIEW

**Review by SkidanovAlex — APPROVED** [review]

File: chain/rosetta-rpc/src/lib.rs [hunk 1]
```diff
@@ -700,14 +700,14 @@ fn get_cors(cors_allowed_origins: &[String]) -> CorsFactory {
 
 pub fn start_rosetta_rpc(
     config: crate::config::RosettaRpcConfig,
-    genesis: Arc<Genesis>,
+    genesis_config: GenesisConfig,
```
Comment (frol) [inline chain/rosetta-rpc/src/lib.rs:703]: Let's keep `Arc<Genesis>` here. Rosetta is optional, and disabled by default, and it is a trade-off we are willing to take for it. Rosetta exposes genesis records as a fake first block, so it needs those records.

File: neard/src/config.rs [hunk 2]
```diff
@@ -533,14 +533,14 @@ pub struct NearConfig {
     #[cfg(feature = "rosetta_rpc")]
     pub rosetta_rpc_config: Option<RosettaRpcConfig>,
     pub telemetry_config: TelemetryConfig,
-    pub genesis: Arc<Genesis>,
+    pub genesis: Genesis,
```
[the remark this answers is not in the document]
Comment (bowenwang1996) [reply neard/src/config.rs:536]: It seems to me that `NearConfig` is not really held in memory. `start_with_config` should drop it at the end. Did I miss something here?

[the remark this answers is not in the document]
Comment (frol) [reply neard/src/config.rs:536]: I missed the fact that it is dropped. If that is the case, we are good

**Comment by frol** [comment]
I don’t believe the code with this change can be compiled with `rosetta_rpc` feature enabled. @chefsale  can we test compilation with all the features enabled? (Add a job to do `cargo check --all-features`)

**Comment by bowenwang1996** [comment]
> I don’t believe the code with this change can be compiled with rosetta_rpc feature enabled. @chefsale can we test compilation with all the features enabled? (Add a job to do cargo check --all-features)

Why do you think so? CI has the following check
```
cargo check --all --tests --benches --all-features
```

**Review by frol — APPROVED** [review]
The plan LGTM, please, make sure that rosetta RPC is compilable.

**Review by nearmax — CHANGES_REQUESTED** [review]
Rosetta still seems to be keeping a copy of the genesis records here: https://github.com/nearprotocol/nearcore/blob/38175236f65b3efa675fb2f68ab0a4a1317cbf39/chain/rosetta-rpc/src/lib.rs#L703

In general, it was difficult to verify that no actor actually stores a copy of the genesis or genesis records for itself locally, since there is a lot of cloning. I am not suggesting to make any changes now, but in general we should start thinking about leveraging the strong-typeness of Rust like me, @frol , and @evgenykuzyakov discussed. For instance in this case we could declare records `!Clone` like here
https://doc.rust-lang.org/src/core/clone.rs.html#234 To make sure the code needs to explicitly opt in into cloning.

**Comment by bowenwang1996** [comment]
@nearmax @frol explicitly requested that we keep this copy when rosetta is enabled https://github.com/nearprotocol/nearcore/pull/3320#discussion_r488666710

**Review by nearmax — APPROVED** [review]

**Pull request merged**